src/base/bittorrent/sessionimpl.cpp

```diff
@@ -3084,27 +3061,21 @@ void SessionImpl::generateResumeData()
 {
     for (TorrentImpl *const torrent : asConst(m_torrents))
     {
-        if (!torrent->isValid()) continue;
-
         if (torrent->needSaveResumeData())
-        {
-            torrent->saveResumeData();
-            m_needSaveResumeDataTorrents.remove(torrent->id());
-        }
+            torrent->requestResumeData();
```

Comment:
I'm a bit too late, but don't we want to pass `lt::torrent_handle::only_if_modified` here too?

Reply:
Although, after testing the change, I see no difference in behavior.

I noticed that all resume files are always overwritten unconditionally by timer. Regardless of torrents being active during this period or not (I refer to `Last Activity` stat).

But turns out libtorrent has different understanding of "modified", or I misunderstand the meaning of the flag here.

Reply:
You could make backup of some fastresume file and then compare it with updated one to see what is changed.

Reply:
Yeah, the files are changing. It is `last_seen_complete` and some other stats.

Btw, reducing the timer to 1 minute makes it visible that the files do not actually update every time. Some updates are delayed for a couple of minutes.
But that happens regardless of the `only_if_modified` flag set. So still seems like it does not change anything.

Reply:
> But that happens regardless of the `only_if_modified` flag set. So still seems like it does not change anything.

Sorry, it isn't clear enough what do you do exactly. If you just modify this line by adding discussed flag then it is meaningless since it is under `if (torrent->needSaveResumeData())` condition.

Reply:
Yes, I kinda suspected that.

I am trying to understand why there is a discrepancy between

https://github.com/qbittorrent/qBittorrent/blob/f87ea1b5d3e0ac6cb3d3a6fef41721b5896b6645/src/base/bittorrent/sessionimpl.cpp#L3064-L3065

and

https://github.com/qbittorrent/qBittorrent/blob/f87ea1b5d3e0ac6cb3d3a6fef41721b5896b6645/src/base/bittorrent/sessionimpl.cpp#L3076-L3080

Which method is better?

Reply:
I mean, I am testing with

```patch
--- a/src/base/bittorrent/sessionimpl.cpp
+++ b/src/base/bittorrent/sessionimpl.cpp
@@ -3061,8 +3061,7 @@ void SessionImpl::generateResumeData()
 {
     for (TorrentImpl *const torrent : asConst(m_torrents))
     {
-        if (torrent->needSaveResumeData())
-            torrent->requestResumeData();
+        torrent->requestResumeData(lt::torrent_handle::only_if_modified);
     }
 }
 
```

and didn't notice any difference.

Which means we can also remove `needSaveResumeData` method completely, as it is not used anywhere else.

```patch
--- a/src/base/bittorrent/torrentimpl.cpp
+++ b/src/base/bittorrent/torrentimpl.cpp
@@ -794,11 +794,6 @@ bool TorrentImpl::connectPeer(const PeerAddress &peerAddress)
     return true;
 }
 
-bool TorrentImpl::needSaveResumeData() const
-{
-    return m_nativeStatus.need_save_resume;
-}
-
 void TorrentImpl::requestResumeData(const lt::resume_data_flags_t flags)
 {
     m_nativeHandle.save_resume_data(flags);

--- a/src/base/bittorrent/torrentimpl.h
+++ b/src/base/bittorrent/torrentimpl.h
@@ -254,8 +254,6 @@ namespace BitTorrent
         void fetchDownloadingPieces(std::function<void (QBitArray)> resultHandler) const override;
         void fetchAvailableFileFractions(std::function<void (QVector<qreal>)> resultHandler) const override;
 
-        bool needSaveResumeData() const;
-
         // Session interface
         lt::torrent_handle nativeHandle() const;
 

```